src/geometry.cpp

```diff
@@ -130,22 +160,9 @@ find_cell_inner(Particle& p, const NeighborList* neighbor_list)
       // Found a material cell which means this is the lowest coord level.
 
       // Find the distribcell instance number.
-      int offset = 0;
       if (c.distribcell_index_ >= 0) {
-        for (int i = 0; i < p.n_coord(); i++) {
-          const auto& c_i {*model::cells[p.coord(i).cell]};
-          if (c_i.type_ == Fill::UNIVERSE) {
-            offset += c_i.offset_[c.distribcell_index_];
-          } else if (c_i.type_ == Fill::LATTICE) {
-            auto& lat {*model::lattices[p.coord(i + 1).lattice]};
-            const auto& i_xyz {p.coord(i + 1).lattice_i};
-            if (lat.are_valid_indices(i_xyz)) {
-              offset += lat.offset(c.distribcell_index_, i_xyz);
-            }
-          }
-        }
+        p.cell_instance() = cell_instance_at_level(p, p.n_coord() - 1);
       }
-      p.cell_instance() = offset;
```

Comment:
What happens if the `if` above is false? Before it looks like we set a cell instance of zero but now we don't set it. Is that intentional?

Reply:
Nope. That's an oversight. Thanks!